ibu-imager/installation_configuration_files/scripts/installation-configuration.sh

```diff
@@ -114,4 +114,36 @@ if [ -d ${EXTRA_MANIFESTS_PATH} ]; then
 fi
 
 rm -rf /opt/openshift
+
+restore_IBU() {
+    K8S_KUBECONFIG_RAW="$(oc config view --raw --minify -ojson)"
+    K8S_KUBECONFIG_CLUSTER="$(jq -r '.clusters[0].cluster' <<< $K8S_KUBECONFIG_RAW)"
+    K8S_KUBECONFIG_USER="$(jq -r '.users[0].user' <<< $K8S_KUBECONFIG_RAW)"
+
+    # Extract the server
+    K8S_API_ENDPOINT="$(jq -r '.server' <<< $K8S_KUBECONFIG_CLUSTER)"
+
+    # Cluster CA certs
+    K8S_API_CA_CERTS=$(jq -r '."certificate-authority-data" | @base64d' <<< $K8S_KUBECONFIG_CLUSTER)
+
+    # User credentials
+    K8S_API_CLIENT_CERT=$(jq -r '."client-certificate-data" | @base64d' <<< $K8S_KUBECONFIG_USER)
+    K8S_API_CLIENT_CERT_KEY=$(jq -r '."client-key-data" | @base64d' <<< $K8S_KUBECONFIG_USER)
+
+    if oc apply -f ${IBU_FILE}; then
+        curl -X PATCH "$K8S_API_ENDPOINT"/apis/lca.openshift.io/v1alpha1/namespaces/openshift-lifecycle-agent/imagebasedupgrades/upgrade/status -H "Content-Type: application/merge-patch+json" -d @${IBU_FILE} --cacert <(cat <<< "$K8S_API_CA_CERTS") --cert <(cat <<< "$K8S_API_CLIENT_CERT") --key <(cat <<< "$K8S_API_CLIENT_CERT_KEY")
```

Comment:
Wouldn't that only update the status? Shouldn't we apply the entire CR?

Reply:
`oc apply` creates the CR. The curl command restores the status.